### PR TITLE
Remove the longest branch that cannot be executed

### PR DESCRIPTION
**Pull request by zhxymh**

### CODE REVIEW

File: src/AElf.Kernel.SmartContractExecution/Application/IBlockExecutingService.cs [hunk 1]
```diff
@@ -115,6 +120,12 @@ await LocalEventBus.PublishAsync(new BlockAcceptedEvent()
             {
                 Logger.LogWarning($"Block validate fails after execution. Exception message {ex.Message}");
             }
+            catch (Exception e)
```
Comment (rosona) [inline src/AElf.Kernel.SmartContractExecution/Application/IBlockExecutingService.cs]: Keep the same naming style for exception

File: src/AElf.Kernel.Core/Blockchain/Domain/IChainManager.cs [hunk 2]
```diff
@@ -410,5 +411,16 @@ private async Task RemoveChainBranchesAsync(Chain chain, List<string> branchKeys
 
             await _chains.SetAsync(chain.Id.ToStorageKey(), chain);
         }
+
+        public async Task RemoveCannotExecutedLongestBranchAsync(Chain chain)
```
Comment (loning) [inline src/AElf.Kernel.Core/Blockchain/Domain/IChainManager.cs]: Can you name it in a OOP way ???

Comment (zhxymh) [reply src/AElf.Kernel.Core/Blockchain/Domain/IChainManager.cs]: How about "RemoveNonExecutableLongestBranchAsync"?

Comment (zhxymh) [reply src/AElf.Kernel.Core/Blockchain/Domain/IChainManager.cs]: > Can you name it in a OOP way ???

Sorry, I got it. Maybe "RemoveLongestBranchAsync" is better?

**Pull request merged**